### PR TITLE
fix(core): fix tab focus, focus styles on popover and avatars, overflow trigger, avatar titles and roles for Avatar Group

### PR DESCRIPTION
**Pull request by olshansky**
**BREAKING CHANGE:**
- Added for `noPadding`, `noHorizontalScroll` and `noVerticalScroll` inputs for `AvatarGroupOverflowBodyDirective` (`fd-avatar-group-overflow-body`).
- Added `AvatarGroupFocusableAvatarDirective` (`fd-avatar-group-focusable-avatar`) needed to bind styles for focusable Avatars.
- Changes in markup

____

**NOTE:** in this PR fundamental-styles updated to `0.18.1-rc.23`,  because of a few issues resolved in styles.

#### Please provide a link to the associated issue.
Closes: https://github.com/SAP/fundamental-ngx/issues/5071
Closes: https://github.com/SAP/fundamental-ngx/issues/5278
Closes: #5047
Closes: https://github.com/SAP/fundamental-ngx/issues/5285

#### Please provide a brief summary of this pull request.
- Group Type tab focusing on the whole Avatar Group and can trigger overflow popover by Enter key
- focus styles on popover-control
- aria-labels for avatars
- total number of team members in case of Group Type

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [n/a] Run npm run build-pack-library and test in external application

Documentation checklist:
- [n/a] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Review by droshev — CHANGES_REQUESTED** [review]
@olshansky this is not a `fix:`, since the changes are in the docs

File: apps/docs/src/app/core/component-docs/avatar-group/examples/avatar-group-group-type-example.component.scss [hunk 1]
```diff
@@ -23,3 +23,10 @@
         }
     }
 }
+
+.fd-avatar-group--group-type__popover-control--custom-class:focus {
+    outline-offset: 0.0625rem;
+    outline-width: 0.0625rem;
+    outline-color: #000;
+    outline-style: dotted;
+}
```
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/avatar-group/examples/avatar-group-group-type-example.component.scss:32]: You can't introduce a class like this in the documentation. 
1st - the naming is absurd!!! Please check the BEM guidelines
2nd - the problem is fixed only in fund-ngx documentation. If the user implements this component the problem will still be there.
3rd - in the documentation scss files you should only have classes that make the rendering of the components nice aka the documentation representation, you should not introduce or modify fundamental classes.

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

**Comment by InnaAtanasova** [comment]
Please use more descriptive name for the PR and provide summary in the PR template.

File: libs/core/src/lib/avatar-group/avatar-group.component.scss [hunk 2]
```diff
@@ -767,3 +767,14 @@
...
     color: var(--fdAvatar_Accent_Color,var(--sapContent_ContrastTextColor,#fff))
 }
+
+// Keep it here after removing styles
+.fd-avatar-group--group-type {
+    fd-popover-control:focus,
+    .fd-popover__control:focus {
+        outline-offset: 0.0625rem;
+        outline-width: 0.0625rem;
+        outline-color: #000;
```
Comment (droshev) [inline libs/core/src/lib/avatar-group/avatar-group.component.scss:777]: why can't we have this in fundamental styles? and why do you use hardcoded color?

**Review by droshev — CHANGES_REQUESTED** [review]
if you are touching the component then the PR should not start with `docs:`

File: libs/core/src/lib/avatar-group/avatar-group.component.scss (hunk 2 above)
Comment (InnaAtanasova) [reply libs/core/src/lib/avatar-group/avatar-group.component.scss:777]: the 3 values are hard-coded. It should be:

```
outline-width: var(--sapContent_FocusWidth);
outline-color: var(--sapContent_FocusColor);
outline-style: var(--sapContent_FocusStyle);
```

Comment (InnaAtanasova) [reply libs/core/src/lib/avatar-group/avatar-group.component.scss:777]: but it should also go in Fundamental-styles

Comment (olshansky) [reply libs/core/src/lib/avatar-group/avatar-group.component.scss]: @droshev @InnaAtanasova thanks, I'll shortly raise an issue in the styles repository to fix it there. 
Now fixed here.

**Review by InnaAtanasova — APPROVED** [review]

File: libs/core/src/lib/avatar-group/avatar-group.component.scss [hunk 3]
```diff
@@ -767,3 +767,14 @@
     color: #fff;
     color: var(--fdAvatar_Accent_Color,var(--sapContent_ContrastTextColor,#fff))
 }
+
```
Comment (droshev) [inline libs/core/src/lib/avatar-group/avatar-group.component.scss:770]: I think it won't a problem if you bring this in styles, merge and then use that rc in your PR

File: apps/docs/src/app/core/component-docs/avatar-group/examples/avatar-group-group-type-example.component.html [hunk 4]
```diff
@@ -1,32 +1,38 @@
...
+    <fd-popover [(isOpen)]="isOpen"
+                [noArrow]="false"
                 [focusTrapped]="true"
                 [focusAutoCapture]="true"
                 [closeOnEscapeKey]="true"
                 (isOpenChange)="isOpenChanged($event)"
                 placement="bottom"
                 additionalBodyClass="fd-avatar-group--group-type__overflow-popover-body--custom-class">
-        <fd-popover-control>
+        <fd-popover-control tabindex="0" (keyup.enter)="toggleOverflow()">
```
Comment (mikerodonnell89) [inline apps/docs/src/app/core/component-docs/avatar-group/examples/avatar-group-group-type-example.component.html:10]: Space should open it as well, as it does for individual type

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]

**Review by mikerodonnell89 — APPROVED** [review]

File: libs/core/src/lib/avatar-group/avatar-group.component.scss (hunk 3 above)
Comment (olshansky) [reply libs/core/src/lib/avatar-group/avatar-group.component.scss:770]: Waiting for https://github.com/SAP/fundamental-styles/pull/2261

File: apps/docs/src/app/core/component-docs/avatar-group/examples/avatar-group-group-type-example.component.ts [hunk 5]
```diff
@@ -17,13 +18,26 @@ export class AvatarGroupGroupTypeExampleComponent {
...
+    isOpen = false;
 
     get isDetailStage(): boolean {
         return this.overflowPopoverStage === 'detail';
     }
 
     constructor(private readonly avatarGroupDataExampleService: AvatarGroupDataExampleService) {}
 
+    handleKeyupOnPopoverControl(event: KeyboardEvent): void {
+        if (!KeyUtil.isKeyCode(event, ENTER) && !KeyUtil.isKeyCode(event, SPACE)) {
```
Comment (Betrozov) [inline apps/docs/src/app/core/component-docs/avatar-group/examples/avatar-group-group-type-example.component.ts:30]:
```suggestion
        if (!KeyUtil.isKeyCode(event, [ENTER, SPACE])) {
```

File: libs/core/src/lib/avatar-group/avatar-group.component.scss (hunk 3 above)
Comment (droshev) [reply libs/core/src/lib/avatar-group/avatar-group.component.scss:770]: approved

Comment (olshansky) [reply libs/core/src/lib/avatar-group/avatar-group.component.scss]: @droshev updated

**Comment by Betrozov** [comment]
@olshansky please resolve conflicts

**Review by Seamoo13 — APPROVED** [review]

**Review by N1XUS — APPROVED** [review]

**Comment by puru-hk** [comment]
Issue tested it is working as expected.

**Review by droshev — APPROVED** [review]

**Pull request merged**